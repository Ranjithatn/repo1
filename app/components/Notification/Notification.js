import React from "react";

// const Notification = ({ message, type, id, onClick, timeOutInMillis }) => {

//   return (
//     <div className={"notification " + type} onClick={() => onClick(id)}>
//       <button className="delete" />
//       {message}
//     </div>
//   );
// };

export class Notification extends React.PureComponent {
  componentDidMount() {
    const { id, timeOutInMillis, onClick } = this.props;

    if (timeOutInMillis) {
      setTimeout(() => {
        onClick(id); //calls requestRemoveNotification
      }, timeOutInMillis);
    }
  }

  render() {
    const { id, message, type, onClick } = this.props;
    // console.log("Notification::this.props",this.props);

    if ( typeof message === 'string' || message instanceof String ) {
      return (
        <div className={"notification " + type} onClick={() => onClick(id)}>
          <button className="delete" />
          {message}
        </div>
      );
    }
    else {
      if ( message && message.Message ) { return null; }
      return (
        <div className={"notification " + type} onClick={() => onClick(id)}>
          <button className="delete" />
          { JSON.stringify(message) }
        </div>
      );
    }

  }
}

export default Notification;
