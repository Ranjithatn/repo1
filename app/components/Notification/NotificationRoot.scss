#notification-root {
  .notification {
    cursor: pointer;
  }
  position: absolute;
  top: 10px;
  right: 10px;
  min-width: 200px;
  z-index: 140;
}
