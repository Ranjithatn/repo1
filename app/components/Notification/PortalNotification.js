import React from 'react';
import ReactDOM from 'react-dom';
import Button from '../Button/Button';

import { store } from '../../index';
import { requestLogout } from "../../actions/auth";
import { push } from 'react-router-redux';
import { translate } from '../../utils/intl';

import LoginPopup from '../LoginForm/LoginPopup';

const PortalNotification = (props) => {

  // console.log("PortalNotification:::props", props);

  const resetNotification = () => {
    return ReactDOM.render(
      null,
      document.getElementById('react-portal-container'),
    );
  }

  if ( ! props.popup ) {
    setTimeout( function() {
  		resetNotification();
    }, props.timeout || 3000);
  }
  // }, 50000);


  let style = {
    background: 'rgba(255,0,0,0.7)',
    padding: 20,
    fontsize: 16,
    position: 'absolute',
    top: '10px',
    right: '10px',
    minwidth: '200px',
    textAlign: 'center',
    zIndex: 50,
    color: '#FFF',
    display: 'flex',
    flexDirection: 'row',
    // left: '50%',
  };

  if ( props.popup ) {
    style.padding = 0;
    style.background="transparent";
  }



  if ( props.style ) { style = {...style, ...props.style} };

  const handleClick = () => {
    resetNotification();
    store.dispatch({ type: 'REQUEST_LOGOUT_SUCCESS' });
    store.dispatch({ type: 'RESET_ENTIRE_STORE' });
    store.dispatch( push('/') );
  }

  const showLoginPopup = () => {
    // resetNotification();
    return ReactDOM.render(
     <PortalNotification message={<LoginPopup />} timeout={9999999*99999999} popup={true} style={{ maxWidth: 400 }} />,
      document.getElementById('react-portal-container'),
    );
  }




  const closeThisNotification = () => {
    resetNotification();
  }

  return (
    <div style={ style }>
      { props.message || 'Some Error Occoured.' }
      { props.redirect && <div style={{ margintop: 10, marginLeft: 20 }}><Button id="jwt-failed" type="submit" className="is-info" onClick={ showLoginPopup } text={ translate("reAuthenticate") } /></div> }
      { ! props.popup &&
        <span onClick={ closeThisNotification }><button className="delete" /></span>
      }
    </div>
  )

}

export const displayError = (message) => {
  return ReactDOM.render(
   <PortalNotification message={message || 'Error Occurred'} timeout={ 7000 } style={{ maxWidth: 400 }} />,
    document.getElementById('react-portal-container'),
  );
}


export const hideError = () => {
  return ReactDOM.render(
    null,
    document.getElementById('react-portal-container'),
  );
}


export const displayLoginPopup = (message) => {
  // console.log("displayLoginPopup called", message);
  // hideError();
  return ReactDOM.render(
   <PortalNotification message={<LoginPopup message={message} />} timeout={9999999*99999999} popup={true} style={{ maxWidth: 400 }} />,
    document.getElementById('react-portal-container'),
  );
}



export default PortalNotification;
