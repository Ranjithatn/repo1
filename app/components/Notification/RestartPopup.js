import React from "react";
import "../LoginForm/LoginPopup.scss";

import Label from "../../components/Label/Label";

import Button from "../../components/Button/Button";

import { translate } from "../../utils/intl";

class RestartPopup extends React.Component {
  constructor() {
    super();

    this.restartApp = this.restartApp.bind(this);
  }

  restartApp() {
    const remote = require("electron").remote;
    remote.app.relaunch();
    remote.app.exit(0);
  }

  render() {
    return (
      <div style={{ width: "100%", margin: "0 auto" }}>
        <div className="popup-bg-blur" />

        <div
          className="component--restart-popup"
          style={{ top: "0 !important;" }}
        >
          <div className="popup-content">
            {this.props.message && (
              <div className="error">{this.props.message}</div>
            )}

            <div>
              <Label
                text={translate(
                  "Settings Saved.Please Restart the Application"
                )}
              />
              <div style={{ textAlign: "center" }}>
                <Button
                  text={translate("restart")}
                  type="button"
                  className="is-primary login-button"
                  onClick={this.restartApp}
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default RestartPopup;
