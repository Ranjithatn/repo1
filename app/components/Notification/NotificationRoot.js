import React, { Fragment } from "react";
import ReactDOM from "react-dom";
import { connect } from "react-redux";
import Notification from "./Notification";
import { requestRemoveNotification } from "../../actions/notifications";
const notificationRoot = document.getElementById("notification-root");
import "./NotificationRoot.scss";

export const NotificationRoot = ({ notifications, requestRemoveNotification }) => {
  if (!notifications || notifications.length <= 0) return <span />;
  return ReactDOM.createPortal(
    <Fragment>
      {notifications.map(notification => {
        return (
          <Notification
            key={notification.id}
            id={notification.id}
            message={notification.message}
            type={notification.type}
            timeOutInMillis={notification.timeOutInMillis}
            onClick={requestRemoveNotification}
          />
        );
      })}
    </Fragment>,
    notificationRoot
  );
};

const mapState = state => ({
  notifications: state.notifications
});
export default connect(mapState, { requestRemoveNotification })(
  NotificationRoot
);
