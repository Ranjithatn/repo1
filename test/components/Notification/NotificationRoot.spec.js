import React from "react";
import toJson from "enzyme-to-json";
import { NotificationRoot } from "../../../app/components/Notification/NotificationRoot";
import Adapter from "enzyme-adapter-react-16";
import { configure, shallow, mount } from "enzyme";


configure({ adapter: new Adapter() });


describe('<NotificationRoot />', () => {

  it("should render without crashing", () => {
    const component = mount(
      <NotificationRoot />
    );
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });

});
