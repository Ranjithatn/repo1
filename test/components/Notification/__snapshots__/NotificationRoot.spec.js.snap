// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotificationRoot /> should render without crashing 1`] = `
<Component>
  <span />
</Component>
`;
